### PR TITLE
Macro feature

### PR DESCRIPTION
**Pull request by pawel-wozniak**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java [hunk 1]
```diff
@@ -10,12 +12,14 @@
 public class DriverManager {
 
 	private Job2dDriver currentDriver = new LoggerDriver();
+	private final Publisher publisher = new Publisher();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java]: Sorry, but adding observer capabilities to DriverManager was not your task. We are still waiting for the solution from a different group. So either, simplify your solution or we will wait for this feature to be delivered.

File: src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import edu.kis.powp.appbase.Application;
+import edu.kis.powp.jobs2d.drivers.decorator.RecordingDriverDecorator;
+import edu.kis.powp.jobs2d.events.SelectRecordMacroMenuOptionListener;
+
+import javax.swing.*;
+import java.awt.event.ActionListener;
+import java.util.Arrays;
+
+public class RecordingFeature {
+	private static Application app;
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java]: It seems not to be used.

**Review by kowallus — CHANGES_REQUESTED** [review]
Thanks. In general, it looks like a good work, but I suggest to approach it differently.

File: src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java (hunk 1 above)
Comment (pawel-wozniak) [reply src/main/java/edu/kis/powp/jobs2d/drivers/DriverManager.java]: I dodn't notice that there is separate story for that. Also, there was a tip in description to use observer pattern. Removed and waiting for implementation.

File: src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java (hunk 2 above)
Comment (pawel-wozniak) [reply src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java]: That is right, removed.

**Comment by pawel-wozniak** [comment]
Moving to blocked category, new PR will be submitted when ready.

**Pull request closed without merging**